pkg/platform/apigateway.go

```diff
@@ -110,3 +139,39 @@ func GetAPIGatewaySensitiveField() []*regexp.Regexp {
 	}
 	return regexpList
 }
+
+func (s *APIGatewayScrubber) ScrubAPIGatewayConfig(ctx context.Context,
+	apiGatewayConfig *APIGatewayConfig) (*APIGatewayConfig, error) {
+	var err error
+	name := apiGatewayConfig.Meta.Name
+	ns := apiGatewayConfig.Meta.Namespace
+
+	existingSecretName, err := s.GetObjectSecretName(ctx, name, ns)
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to get api gateway config secret name")
+	}
```

Comment:
is that necessary?
You get the secret again in `GetExistingSecretAndScrub`

Reply:
I've optimised this logic, so solving the comment